synapse/events/utils.py

```diff
@@ -475,14 +475,15 @@ def serialize_event(
     if config.as_client_event:
         d = config.event_format(d)
 
-    # If the event is a redaction, copy the redacts field from the content to
-    # top-level for backwards compatibility.
-    if (
-        e.type == EventTypes.Redaction
-        and e.room_version.updated_redaction_rules
-        and e.redacts is not None
-    ):
-        d["redacts"] = e.redacts
+    # If the event is a redaction, either copy the redacts field from the content to
+    # top-level for backwards compatibility OR from the redacts field to the content
+    # field for forwards compatibility.
+    if e.type == EventTypes.Redaction and e.redacts is not None:
+        if e.room_version.updated_redaction_rules:
+            d["redacts"] = e.redacts
+        else:
+            d["content"] = dict(d["content"])
+            d["content"]["redacts"] = e.redacts
```

Comment:
This needs clarification of what to do if the property exists, but is different.

E.g. if an event of the form below is received:

```json5
{
    "sender": "@alice:example.com",
    // ... other fields
    "content": {
        "redacts": "$abcdef"
    },
    "redacts": $qrstuv"
}
```

Right now this would just stomp on the fields "correcting" them, which seems OK, but not really great. I'm tempted to say we should rip out the backwards compatibility completely and force clients to look in the different spots for the redacts field based on the room version.

Any thoughts?

Reply:
From further conversation with @turt2live, overwriting it here is the proper thing to do.